index.js

```diff
@@ -149,42 +191,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let num = 0
+function takeATicketNumber(line){
+  num++
+  line.push(num)
+  return `Welcome. You are ticket number ${num}`
+}
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
 
 // ***** Scope & Closures - Question 2 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+
+function ticketNumberGeneratorFunc(){
+  number = 0
+  function takeATicketNumberFunc(array){
+    number++
+    array.push(number)
+    return `Welcome. You are ticket number ${number}`
+  }
+  return takeATicketNumberFunc
+}
```

Comment:
Nice work on this! Just be careful on line 229 here, if you declare a variable without `let` or `const` it will become a global variable (even if you declare it inside a function). To make sure that the ticket variable here is scoped to the function, use `let`.